index.js

```diff
@@ -26,7 +26,7 @@ const defaultOpts = {
   useEval: false,
   // A RegExp to find `new Workers()` calls. The second capture group _must_
   // capture the provided file name without the quotes.
-  workerRegexp: /new Worker\((["'])(.+?)\1(,[^)]+)?\)/g,
+  workerRegexp: /new Worker\((.+?)(,[^)]+)?\)/g,
```

Comment:
This will fail on any expression with parentheses, e.g. for:

`new Worker(new URL("some-url", import.meta.url), {type:'module'})`

it will match:

`new Worker(new URL("some-url", import.meta.url)`

stopping on the first closing paren.

I understand you're using regexes as an optimisation, but perhaps it's worth doing something like `if (code.includes('new Worker')) { let ast = this.parse(code); ... }`?

It would still quickly skip files that don't have Worker constructors altogether, but will handle comments and parentheses more gracefully.